components/nimbus/src/enrollment.rs

```diff
@@ -2472,6 +2480,69 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_evolver_opt_in() -> Result<()> {
+        let existing_enrollments = vec![
```

Comment:
The tests pass even without the change, which is mildly concerning. This is because of what I noted above, we end up returning the `OptIn` enrollment even though we evaluate the enrollment.